### PR TITLE
Snakemake hpcc dev

### PR DESCRIPTION
**Pull request by armankhoshnevis**
Here is problem: How can I ensure that jobs are being submitted as SLURM?

### CODE REVIEW

File: workflow/Snakefile [hunk 1]
```diff
@@ -18,7 +20,11 @@ rule run_pso_optimization_fmg:
         matlab_function = "main_loop_hpcc_function",
     log:
         "logs/run_pso_fmg_{metadata}.log"
+    resources:
+        time=config["default-resources"]["time"],
+        mem=config["default-resources"]["mem"],
+        cpu_per_task=config["default-resources"]["cpus_per_task"]
```
Comment (andrewfullard) [inline workflow/Snakefile]: This shouldn't be necessary- the point of default resources is they are the default for all rules

**Comment by andrewfullard** [comment]
Remember, to get slurm to be used, you need to install the slurm executor plugin before running snakemake. The slurm profile should be loaded like this:
`snakemake --workflow-profile slurm` see e.g. https://msu-cmse-courses.github.io/CMSE_890-602_snakemake/workshop_material/04_leveling_up_your_workflow/#41-use-a-profile-for-hpc

**Comment by armankhoshnevis** [comment]
> Remember, to get slurm to be used, you need to install the slurm executor plugin before running snakemake. The slurm profile should be loaded like this: `snakemake --workflow-profile slurm` see e.g. https://msu-cmse-courses.github.io/CMSE_890-602_snakemake/workshop_material/04_leveling_up_your_workflow/#41-use-a-profile-for-hpc

Thanks for your comments.

The command I use for running the workflow is as follows:
snakemake --workflow-profile slurm

As far as I can learn from the link you provided, the config file looks okay default-resources and executor are defined properly.

However, I got the DUE TO TIME LIMIT error for all jobs. Moreover, the following message appears right after the running snakemake command in the command line:
No SLURM account given, trying to guess.
Guessed SLURM account: general
No partition was given for rule 'run_pso_optimization_fmg', and unable to find a default partition. Trying to submit without partition information. You may want to invoke snakemake with --default-resources 'slurm_partition=<your default partition>'.
No wall time information given. This might or might not work on your cluster. If not, specify the resource runtime in your rule or as a reasonable default via --default-resources.

This is immediate looks very strange as I have clearly specified cluster resources.

I am so confused :)

**Comment by andrewfullard** [comment]
I had a closer look at your config.yaml and it does not seem to follow the correct format. For example, instead of `time: 00:01:00` for one hour, the entry should be `runtime: 60` for 60 minutes. You can see the documentation here: https://snakemake.github.io/snakemake-plugin-catalog/plugins/executor/slurm.html#advanced-resource-specifications which lists how SLURM settings map to the configuration.

**Comment by armankhoshnevis** [comment]
> I had a closer look at your config.yaml and it does not seem to follow the correct format. For example, instead of `time: 00:01:00` for one hour, the entry should be `runtime: 60` for 60 minutes. You can see the documentation here: https://snakemake.github.io/snakemake-plugin-catalog/plugins/executor/slurm.html#advanced-resource-specifications which lists how SLURM settings map to the configuration.

Thank you for your feedback.

The workflow is working properly on the HPCC :)

**Comment by armankhoshnevis** [comment]
mergepr

**Pull request merged**